ninja-core/src/test/java/ninja/utils/LambdasTest.java

```diff
@@ -0,0 +1,197 @@
+        String value = (String)lambdaInfo.getFunctionalMethod().invoke(lambda, 1L);
+        //String value = (String)lambda.getClass().getMethod("apply", Long.class).invoke(calc, 1L);
+        
+        assertThat(value, is("2"));
+    }
+    
+    private Result home() {
+        return Results.html().renderRaw("Hi".getBytes(StandardCharsets.UTF_8));
+    }
+    
+    @Test
+    public void anyInstanceMethodReference() throws Exception {
+        ControllerMethod1<LambdasTest> lambda = LambdasTest::home;
+        
+        LambdaInfo lambdaInfo = Lambdas.reflect(lambda);
+        
+        assertThat(lambdaInfo.getKind(), is(Kind.ANY_INSTANCE_METHOD_REFERENCE));
+        
+        SerializedLambda serializedLambda = lambdaInfo.getSerializedLambda();
+        
+        assertThat(serializedLambda.getFunctionalInterfaceMethodName(), is("apply"));
+        assertThat(serializedLambda.getImplClass().replace('/', '.'), is(LambdasTest.class.getCanonicalName()));
+        assertThat(serializedLambda.getImplMethodName(), is("home"));
+        assertThat(serializedLambda.getImplMethodSignature(), is("()Lninja/Result;"));
+        assertThat(serializedLambda.getCapturedArgCount(), is(0));
+    }
+    
+    @Test
+    public void anonymousClassReference() throws Exception {
+        @SuppressWarnings("Convert2Lambda")
+        ControllerMethod1<Context> lambda = new ControllerMethod1<Context>() {
+            @Override
+            public Result apply(Context a) {
+                return Results.html().renderRaw("".getBytes(StandardCharsets.UTF_8));
+            }
+        };
+        
+        try {
+            LambdaInfo lambdaInfo = Lambdas.reflect(lambda);
+            fail();
+        } catch (IllegalArgumentException e) {
+            // expected
+        }
+    }
+    
+    @Test
+    public void anonymousMethodReference() throws Exception {
+        ControllerMethod1<Context> lambda = (Context context) -> Results.html().renderRaw("".getBytes(StandardCharsets.UTF_8));
+        
+        LambdaInfo lambdaInfo = Lambdas.reflect(lambda);
+        
+        assertThat(lambdaInfo.getKind(), is(Kind.ANONYMOUS_METHOD_REFERENCE));
+        
+        SerializedLambda serializedLambda = lambdaInfo.getSerializedLambda();
+        
+        assertThat(serializedLambda.getFunctionalInterfaceMethodName(), is("apply"));
+        assertThat(serializedLambda.getImplClass().replace('/', '.'), is(LambdasTest.class.getCanonicalName()));
+        assertThat(serializedLambda.getImplMethodName(), startsWith("lambda$"));
+        assertThat(serializedLambda.getInstantiatedMethodType(), is("(Lninja/Context;)Lninja/Result;"));
+        // includes captured args btw...
+        assertThat(serializedLambda.getImplMethodSignature(), is("(Lninja/Context;)Lninja/Result;"));
+        assertThat(serializedLambda.getImplMethodKind(), is(6));    // 6 = REF_invokeStatic
+        assertThat(serializedLambda.getCapturedArgCount(), is(0));
+    }
+    
+    @Test @Ignore
```

Comment:
Deleted.  Was using to prove how anonymous lambdas do not honor parameter annotations.  However, its not actually needed.